src/lib/run_environment.ml

```diff
@@ -94,10 +94,37 @@ module Tool = struct
   end
 end
 
+module Make_fun = struct
+  module Requirement = struct
+    type t = [
+      | `Processors of int
+      | `Internet_access
+      | `Memory of [ `GB of float | `Decent | `Big ]
+      | `Quick_run
+      | `Spark of string list (* custom parameters *)
+      | `Custom of string
+    ] [@@deriving yojson, show]
+  end
```

Comment:
1. It's exported; users of the library have to match on it.
2. Also within the library, we have to put/compute actual literals
3. Polymorphic variants allow nice nesting (like `` | `Memory of [ `GB of float | `Decent | `Big ]``
4. No good reason to use regular variants (the performance difference is meaningless)